python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -53,6 +54,19 @@
 from .types import ExecuteExternalPipelineArgs
 
 
+class ErrorOnMissingInstance(DynamicPartitionsStore):
```

Comment:
this kind of exploding class was my original proposal - but now that I know most of the code is able to handle an empty instance my proposal was to modify this error message instead to have this text https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/definitions/partition.py#L614-L618